src/TranslatesFactoryData.php

```diff
@@ -11,16 +11,30 @@ private static function isFactory($item): bool
         return $item instanceof BaseFactory || $item instanceof FactoryBuilder;
     }
 
-    private function transformFactoriesToRelationIds(array $defaultModelFields): array
+    private static function isCallable($field): bool
     {
-        return collect($defaultModelFields)
-            ->map(function ($item) {
-                if ($this->isFactory($item)) {
-                    return $item->create()->getKey();
-                }
-
-                return $item;
-            })
-            ->toArray();
+        return is_callable($field) && ! is_string($field) && ! is_array($field);
+    }
+
+    private function transformModelFields(array $defaultModelFields): array
+    {
+        foreach ($defaultModelFields as &$field) {
+            $field = $this->transformField($field, $defaultModelFields);
+        }
+
+        return $defaultModelFields;
+    }
+
+    private function transformField($field, array $defaultModelFields)
+    {
+        if ($this->isFactory($field)) {
+            return $field->create()->getKey();
+        }
+
+        if ($this->isCallable($field)) {
+            return $field($defaultModelFields);
+        }
+
+        return $field;
```

Comment:
This portion was heavily inspired by how Laravel's `FactoryBuilder@expandAttributes` method works. 

The main takeaway here is that we use foreach with a variable passed by reference so with each iteration, the individual field transforms into the closure's return value, thus making it usable in the next factory fields. With map, the fields aren't affected until it loops all the way through. 

I can provide screenshots if i didn't explain this part well.

Reply:
yeah, i think my test wasnt testing the actual problem with using map. I have updated the case in 627161fbb5e960b4beaa61e3c1bca57e2b16e7ab

The problem is when i define an array of defaults, for example: 
```php
public function getDefaults(Generator $faker): array
{
  return [
    'first_field' => fn (array $factory) => 'some string',
    'second_field' => fn(array $factory) => someOtherAction($factory['second_field']),
  ];
}
```

In the array above each key's value is a function. When we loop through these one at a time and transform them to the callbacks return value, the `first_field` key will turn into a string.

When we get to the second field, it's relying on `$factory['second_field']` to already be a string, not a closure.

When we use `->map` to iterate, we get to the `first_field`, we transform it into a string, but don't update it yet. Instead, it holds in a different memory location and will switch at the end. Then, when we get to the `second_field` key, it throws an error because `someOtherAction` requires a string, but a closure is passed because the first field hasn't been transformed yet.

using `foreach` with passing `$field` by reference via `&$field`, it allows the value to transform on each iteration, rather than waiting to switch the entire array at once to the new array of values.

Here is a printout of what the `$defaultModelFields` looks like on each iteration using map and foreach:

**Using Map**
```php

// first iteration
array:2 [
  "name" => Closure(array $ingredient) {#1354
    class: "Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest"
    this: Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest {#759 …}
  }
  "description" => Closure(array $ingredient) {#1355
    class: "Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest"
    this: Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest {#759 …}
  }
]

// second iteration
array:2 [
  "name" => Closure(array $ingredient) {#1354
    class: "Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest"
    this: Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest {#759 …}
  }
  "description" => Closure(array $ingredient) {#1355
    class: "Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest"
    this: Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest {#759 …}
  }
]
```


**Using Foreach**

```php

// first iteration
array:2 [
  "name" => & Closure(array $ingredient) {#1354
    class: "Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest"
    this: Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest {#759 …}
  }
  "description" => Closure(array $ingredient) {#1355
    class: "Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest"
    this: Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest {#759 …}
  }
]

// second iteration
array:2 [
  "name" => "Basil"
  "description" => & Closure(array $ingredient) {#1355
    class: "Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest"
    this: Christophrumpel\LaravelFactoriesReloaded\Tests\FactoryTest {#759 …}
  }
]
```

Reply:
Ah got it. Thanks for the details. Then I'm fine with the current implementation 👍